--- src/JobWaitTrait.php	2023-11-01 13:07:13.826212516 +0100
+++ src/JobWaitTrait.php	2023-11-01 13:07:24.542742533 +0100
@@ -55,7 +55,17 @@
                 }
             };

-            (new ExponentialBackoff($maxRetries))
+            (new ExponentialBackoff($maxRetries, function (\Throwable $e) {
+                if ($e->getCode() === 499) {
+                    // If the job is cancelled, don't retry.
+                    return false;
+                }
+                if ($e->getCode() === 400) {
+                    // If the job is ended with bad request, don't retry.
+                    return false;
+                }
+                return true;
+            }))
                 ->execute($retryFn);
         }
     }
